recipe/meta.yaml

```diff
@@ -21,18 +22,24 @@ requirements:
     - python >=3.7
     - geopandas-base >=0.10
     - dask-core >=2021.06.0
-    - pygeos
```

Comment:
I think we still need to add an explicit `shapely >= 2`, to ensure users have the equivalent of pygeos (it really makes no sense to use dask-geopandas with older shapely, and actually even will error for certain operations)

This might mean we have to remove the pip-check for now, but I think the above is more important.

Reply:
We can add a `run_constrained` (see https://conda-forge.org/docs/maintainer/adding_pkgs.html#constraining-packages-at-runtime), which means shapely won't be pulled in as an explicit runtime dependency, but anyone trying to install shapely alongside will have to pull in `shapely>=2`.

Reply:
For my understanding, what's the advantage of using run_constrained instead of just adding it as an explicit runtime dependency?

> but anyone trying to install shapely alongside will have to pull in shapely>=2.

That also works if shapely is already installed in the env?

Reply:
And to be clear, shapely is not an "optional" dependency (since the link about run_constrained mentions that as a use case). Shapely _is_ a required dependency of dask-geopandas (and also geopandas already has that dependency as well). It's just that in dask-geopandas itself we still allow you to have an older shapely if pygeos is installed (hence, the setup.py of dask-geopandas doesn't mention shapely>2 to allow this flexibility). But here if we don't add pygeos as dependency, that essentially makes shapely > 2 as a required dependency

Reply:
Since `dask-geopandas` depends on `geopandas-base` , which itself has a dependency on `shapely>=1.7.1`, there is no need to declare shapely as a runtime dependency here. I'm thinking we can leave out the `run_constrained` too then, and leave it to the user to manually install either `pygeos` or `shapely>=2.0`. Any constraint we add here on conda-forge will restrict users one way or another, and it's hard to enforce the `if not pygeos, install shapely 2.0` logic as far as I'm aware.